src/blocks/content-helper/current-post-details/component.tsx

```diff
@@ -0,0 +1,309 @@
+			</span>
+		</div>
+	);
+}
+
+/**
+ * Outputs the "General Performance" (Views, Visitors, Time) section.
+ *
+ * @param {PostPerformanceData} data The data needed to populate the section.
+ */
+function GeneralPerformanceSection( data: PostPerformanceData ) {
+	return (
+		<div className="section general-performance">
+			<table>
+				<tbody>
+					<tr>
+						<td>{ ImpreciseNumber( data.views ) }</td>
+						<td>{ ImpreciseNumber( data.visitors ) }</td>
+						<td>{ data.avgEngaged }</td>
+					</tr>
+				</tbody>
+				<tfoot>
+					<tr>
+						<th>{ __( 'Page Views', 'wp-parsely' ) }</th>
+						<th>{ __( 'Visitors', 'wp-parsely' ) }</th>
+						<th title={
+							__( 'Average time visitors engaged with content', 'wp-parsely' )
+						}>{ __( 'Engaged Time', 'wp-parsely' ) }</th>
+					</tr>
+				</tfoot>
+			</table>
+		</div>
+	);
+}
+
+/**
+ * Outputs the "Referrer Types" section.
+ *
+ * @param {PostPerformanceData} data The data needed to populate the section.
+ */
+function ReferrerTypesSection( data: PostPerformanceData ) {
+	// Remove unneeded totals to simplify upcoming map() calls.
+	delete data.referrers.types[ 'totals' as unknown as number ];
+
+	return (
+		<div className="section referrer-types">
+			<div className="section-title">{ __( 'Referrers (Page Views)', 'wp-parsely' ) }</div>
+
+			<div className="multi-percentage-bar">{
+				Object.entries( data.referrers.types ).map( ( [ key, value ] ) => {
+					return (
+						<div className={ 'bar-fill ' + key } key={ key }
+							title={ value.viewsPercentage + __( '%', 'wp-parsely' ) }
+							style={ { width: value.viewsPercentage + '%' } }>
+						</div>
+					);
+				} ) }
+			</div>
+
+			<table>
+				<thead>
+					<tr>{
+						Object.keys( data.referrers.types ).map( ( key ) => {
+							return <th key={ key }>{ key }</th>;
+						} ) }
+					</tr>
+				</thead>
+				<tbody>
+					<tr>{
+						Object.entries( data.referrers.types ).map( ( [ key, value ] ) => {
+							return <td key={ key }>{ ImpreciseNumber( value.views ) }</td>;
+						} ) }
+					</tr>
+				</tbody>
+			</table>
+		</div>
+	);
+}
+
+/**
+ * Outputs the "Top Referrers" section.
+ *
+ * @param {PostPerformanceData} data The data needed to populate the section.
+ */
+function TopReferrersSection( data: PostPerformanceData ) {
+	let totalViewsPercentage = 0;
+	return (
+		<div className="section top-referrers">
+			<table>
+				<thead>
+					<tr>
+						<th>{ __( 'Top Referrers', 'wp-parsely' ) }</th>
+						<th colSpan={ 2 }>{ __( 'Page Views', 'wp-parsely' ) }</th>
+					</tr>
+				</thead>
+				<tbody>{
+					Object.entries( data.referrers.top ).map( ( [ key, value ] ) => {
+						if ( key === 'totals' ) {
+							totalViewsPercentage = Number( value.viewsPercentage );
+							return null;
+						}
+
+						return (
+							<tr key={ key }>
+								<th title={ key }>{ key }</th>
+								<td title={ value.viewsPercentage + __( '%', 'wp-parsely' ) }>
```

Comment:
The number should be added as a placeholder / interpolated value within the string, as for some locales, the percent sign comes before the number.
(This will apply to anywhere that a percent sign is being displayed)

You'll need to import and use `sprintf` from `@wordpress/i18n`.